nni/retiarii/utils.py

```diff
@@ -107,5 +111,46 @@ def top(cls, key: str) -> Any:
         return cls._stack[key][-1]
 
 
+class ModelNamespace:
+    """
+    To create an individual namespace for models to enable automatic numbering.
+    """
+
+    def __init__(self, key: str = _DEFAULT_MODEL_NAMESPACE):
+        # for example, key: "model_wrapper"
+        self.key = key
+
+    def __enter__(self):
+        # For example, currently the top of stack is [1, 2, 2], and [1, 2, 2, 3] is used,
+        # the next thing up is [1, 2, 2, 4].
+        # `reset_uid` to count from zero for "model_wrapper_1_2_2_4"
```

Comment:
what is the relation between this label and user provided label?

Reply:
If user provides a label, it will always override this label.